Sources/PandaSDK/ConfiguredPanda.swift

```diff
@@ -160,14 +160,17 @@ final public class Panda: PandaProtocol {
 
     public func handleApplication(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any]) {
         /// appid://panda/promo/product_id
-        if url.host == "panda",
-            let _ = url.pathComponents.last {
+        if url.host == "panda" {
             
             /// track analytics
             trackDeepLink(url.absoluteString)
             
             /// show screen
-            showScreen(screenType: .promo)
+            if let product = url.pathComponents.last {
+                showScreen(screenType: .product, product: product)
+            } else {
+                showScreen(screenType: .promo)
```

Comment:
invalid logic. need additional validation for `product` or `promo` deep links instead of `url.pathComponents.last`
because in case `appid://panda/promo/` -> `url.pathComponents.last` == `promo` ;(